R/tm_g_bivariate.R

```diff
@@ -508,48 +514,42 @@ srv_g_bivariate <- function(id,
         }
       }
 
-      teal.code::chunks_push(
-        id = "plot_call",
-        expression = substitute(expr = p <- cl, env = list(cl = cl))
-      )
-
       # Add labels to facets
       nulled_row_facet_name <- varname_w_label(row_facet_name, ANL)
       nulled_col_facet_name <- varname_w_label(col_facet_name, ANL)
+      without_facet <- (is.null(nulled_row_facet_name) && is.null(nulled_col_facet_name)) || !facetting
 
-      if ((is.null(nulled_row_facet_name) && is.null(nulled_col_facet_name)) || !facetting) {
-        teal.code::chunks_push(id = "print_plot_call", expression = quote(print(p)))
-        teal.code::chunks_safe_eval()
+      print_call <- if (without_facet) {
+        quote(print(p))
       } else {
-        teal.code::chunks_push(
-          id = "grid_draw_call",
-          substitute(
-            expr = {
-              # Add facetting labels
-              # optional: grid.newpage() #nolintr
-              g <- add_facet_labels(p, xfacet_label = nulled_col_facet_name, yfacet_label = nulled_row_facet_name)
-              grid::grid.draw(g)
-            },
-            env = list(nulled_col_facet_name = nulled_col_facet_name, nulled_row_facet_name = nulled_row_facet_name)
-          )
+        substitute(
+          expr = {
+            # Add facetting labels
+            # optional: grid.newpage() #nolintr
+            p <- add_facet_labels(p, xfacet_label = nulled_col_facet_name, yfacet_label = nulled_row_facet_name)
+            grid::grid.draw(p)
+          },
+          env = list(nulled_col_facet_name = nulled_col_facet_name, nulled_row_facet_name = nulled_row_facet_name)
         )
-        teal.code::chunks_safe_eval()
-        teal.code::chunks_get_var("g")
       }
+
+      teal.code::eval_code(merged$anl_q_r(), substitute(expr = p <- cl, env = list(cl = cl)), name = "plot_call") %>%
+        teal.code::eval_code(print_call, name = "print_call")
```

Comment:
### Explanation comment
previously code has been added to the chunks using `chunks_push` and evaluated at the end using `chunks_safe_eval`.

Now we `eval_code` which is equivalent of `chunks_push + chunks_eval`